backend/src/main/java/de/quizmasters/backend/models/QuizService.java

```
@@ -25,11 +25,19 @@ public Quiz addQuiz(Quiz newQuiz) {
     public Quiz updateQuiz(String id, Quiz updatedQuiz) {
         Optional<Quiz> quizToUpdate = getQuizzes().stream().filter(quiz -> id.equals(quiz.getId())).findFirst();
         if (quizToUpdate.isPresent()) {
-            Quiz quiz = quizToUpdate.get();
-           // quiz = updatedQuiz;
-            quiz.setQuestion(updatedQuiz.getQuestion());
-           quiz.setAnswer(updatedQuiz.getAnswer());
-            return quiz;
+           int index = getQuizzes().indexOf(quizToUpdate.get());
+            getQuizzes().set(index, updatedQuiz);
+            return updatedQuiz;
+        } else {
+            throw new NullPointerException("Quiz not found");
```

NullPointerException is weird, you use optionals to handle null messages, but also want to throw errors when there is a null? I think you wanted to throw a different Error instead of NullPointerException